### PR TITLE
RN 2018.09

### PR DESCRIPTION
**Pull request by mystiberry-okta**
## Description:
- change log for 2018.08+09 This JIRA includes small doc changes that correspond to two of the change log items.

### Resolves:
[OKTA-159656](https://oktainc.atlassian.net/browse/OKTA-159656)

### CODE REVIEW

File: _source/_change-log/2018-09.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+The following changes are available starting February 28, 2018 in preview orgs and starting March 5, 2018 in production orgs:
+
+| [User Consent for OAuth 2.0 and OpenID Connect Flows in Early Availability](#user-consent-for-oauth-20-and-openid-connect-flows-in-early-availability)
+| [Sessions API Supports HTTP Header Prefer](#sessions-api-supports-http-header-prefer)
+| [Improved Response Mode](#improved-response-mode)
+| [Change to `/authorize` Response for `prompt`](#change-to-authorize-response-for-prompt)
+| [Changes to Error Messages from User Creation Requests](#changes-to-error-messages-from-user-creation-requests)
+| [Improved System Log Behavior for Date Queries](#improved-system-log-behavior-for-date-queries)
+| [System Log Message Changes](#system-log-message-changes)
+| [Bug Fixes](#bug-fixes)
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: We need to make this link more specific. Currently, we jump to the `Bug Fixes` title in 2018.02

File: _source/_change-log/2018-09.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+Consent grants are different from tokens because a consent can outlast a token, and there can be multiple tokens with varying sets of scopes derived from a single consent. When an application comes back and needs to get a new access token, it may not need to prompt the user for consent if they have already consented to the specified scopes. Consent grants remain valid until the user manually revokes them, or until the user, application, authorization server or scope is deactivated or deleted.
+
+To configure an authorization or authentication flow:
+
+1. Verify that you have the API Access Management feature enabled, and request that User Consent also be enabled.
+2. Create an app via the Apps API with the appropriate values for `tos_uri`, `policy_uri`, and `consent_method`. [Doc](/docs/api/resources/apps#settings-7). You can also configure an existing app in the Admin user interface: **Applications > [Application Name] > General > User Consent**.
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Is [Doc] alone a sentence?

File: _source/_change-log/2018-09.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+3. Ensure that your authentication or authorization flow is configured properly. The combination of `prompt` in the `/authorize` request, `consent_method` set on the app in the previous step, and `consent`, a property set on scopes, controls whether a user consent window is displayed during the authentication flow. [Details](/docs/api/resources/apps#settings-7)
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Same comment above about [Details]. Seems odd to me that we have these one-word "sentences"

File: _source/_change-log/2018-09.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+<!-- OKTA-158107 -->
+
+### Sessions API Supports HTTP Header Prefer
+
+Okta now supports [the HTTP Header `Prefer`](https://tools.ietf.org/html/rfc7240#section-4.2) in [the Sessions API for refreshing sessions](/docs/api/resources/sessions#refresh-current-session). You can extend the session lifetime, but skip any processing work related to building the response body.
+
+Example Request
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Is there a reason why these are not headers like in our API documentation?

File: _source/_change-log/2018-09.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    "errorSummary": "Api validation failed: com.saasure.core.services.user.InvalidUserProfileException: Could not create user due to invalid profile: com.saasure.framework.validation.util.SimpleErrors: 1 errors\nError in object 'newUser': codes [password.passwordRequirementsNotMet.newUser,password.passwordRequirementsNotMet]; arguments [Password requirements: at least 8 characters, a lowercase letter, an uppercase letter, a number, no parts of your username.]; default message [Password requirements were not met. Password requirements: at least 8 characters, a lowercase letter, an uppercase letter, a number, no parts of your username.]",
+    "errorLink": "E0000001",
+    "errorId": "oaecNfS38enQ8KtWDvNfusWRw",
+    "errorCauses": [
+        {
+            "errorSummary": "Password requirements were not met. Password requirements: at least 8 characters, a lowercase letter, an uppercase letter, a number, no parts of your username."
+        }
+    ]
+}
+------
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: We can remove this line: `-----`

File: _source/_change-log/2018-09.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            "reason": "NUMBER_REQUIRED",
+            "location": "credentials.password.value",
+            "locationType": "body",
+            "domain": "user"
+        }
+    ]
+}
+~~~
+
+If you don't want these changes, contact Support to opt out.
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Maybe add a `mailto:` link here?

File: _source/_change-log/2018-09.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+    * `app.oauth2.authorize.implicit.access_token`
+
+2. The existing message `app.oauth2.as.authorize.implicit` replaces:
+
+    * `app.oauth2.as.authorize.implicit.id_token`
+    * `app.oauth2.as.authorize.implicit.access_token` <!-- OKTA-155402 -->
+
+### Bugs Fixed
+
+The following bugs have been fixed and are expected in preview orgs February 14, 2018 and production orgs starting February 27, 2018.
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Just curious - Are we just late getting out this notice or are the dates incorrect?

File: _source/_change-log/2018-09.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+Example Request
+
+~~~http
+Accept: application/json
+Content-Type: application/json
+Authorization: SSWS ${api_token} or {sid_cookie}
+Prefer: return=minimal
+
+POST https://${yourDomain}.com/api/v1/sessions/{id-or-me}/refresh
```
[the remark this answers is not in the document]
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: ack

File: _source/_change-log/2018-09.md (hunk 3 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: will remove, thanks!

File: _source/_change-log/2018-09.md (hunk 6 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Is that how people reach out to Support? I thought they opened a case...

File: _source/_change-log/2018-09.md (hunk 7 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Nope, wrong dates. Thanks!

File: _source/_change-log/2018-09.md (hunk 4 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Originally I had these so deeply nested I couldn't, but fixed that so adding header markings. Thanks!

File: _source/_change-log/2018-09.md (hunk 5 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: ack

File: _source/_change-log/2018-09.md (hunk 6 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: This is how we say it in doc currently, will create JIRA to investigate and maybe change to better instruction.

Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: https://oktainc.atlassian.net/browse/OKTA-160463

File: _source/_change-log/2018-09.md (hunk 1 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: It did do this (I was looking at the previous link, sorry). @jakubvul we need to include the release number for these repeated headings (or if you have a better solution..)

File: _source/_change-log/2018-09.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+#### Example Request
+
+~~~http
+Accept: application/json
+Content-Type: application/json
+Authorization: SSWS ${api_token} or {sid_cookie}
+Prefer: return=minimal
+
+POST https://{yourDomain}.com/api/v1/sessions/{id-or-me}/refresh
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Looks like we forgot to update the URL here.

File: _source/_change-log/2018-09.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
+---
+layout: docs_page
+title: Okta API Products Change Log
+---
+
+## 2018.09
+
+| Change | Expected in Preview Orgs | Rollout to Production Orgs Expected to Start |
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: Do these dates need to be update too? A month+ rolllout seems odd.

File: _source/_change-log/2018-09.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+| [Sessions API Supports HTTP Header Prefer](#sessions-api-supports-http-header-prefer)  | February 28, 2018               | March 5, 2018 |
+| [Improved Response Mode for OAuth 2.0 and OpenID Connect Requests](#improved-response-mode-for-oauth-20-and-openid-connect-requests)  | February 28, 2018               | March 5, 2018 |
+| [Change to `/authorize` Response for `prompt` for OAuth 2.0 and OpenID Connect Requests](#change-to-authorize-response-for-prompt-for-oauth-20-and-openid-connect-requests)  | February 28, 2018               | March 5, 2018 |
+| [Changes to Error Messages from User Creation Requests](#changes-to-error-messages-from-user-creation-requests)  | February 28, 2018               | March 5, 2018 |
+| [Improved System Log Behavior for Date Queries](#improved-system-log-behavior-for-date-queries)  | February 28, 2018               | March 5, 2018 |
+| [System Log Message Changes Related to Authorization Servers](#system-log-message-changes-related-to-authorization-servers)  | February 28, 2018               | March 5, 2018 |
+| [Bug Fixes for 2018.09](#bug-fixes-for-201809)  | February 28, 2018               | March 5, 2018 |
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: This is a dead link. We'll want to update the `Bug Fixes` header below to match this link.

File: _source/_change-log/2018-09.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+<!-- OKTA-158107 -->
+
+### Sessions API Supports HTTP Header Prefer
+
+Okta now supports [the HTTP Header `Prefer`](https://tools.ietf.org/html/rfc7240#section-4.2) in [the Sessions API for refreshing sessions](/docs/api/resources/sessions#refresh-current-session). You can extend the session lifetime, but skip any processing work related to building the response body.
+
+#### Example Request
+
+~~~http
+Accept: application/json
```
Comment (jmelberg-okta) [inline _source/_change-log/2018-09.md]: nit: In other examples we use entire `curl` requests as opposed to `http`.

File: _source/_change-log/2018-09.md (hunk 9 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Ack.

File: _source/_change-log/2018-09.md (hunk 12 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: fixed.

File: _source/_change-log/2018-09.md (hunk 11 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Fixing, thanks!

File: _source/_change-log/2018-09.md (hunk 10 above)
Comment (mystiberry-okta) [reply _source/_change-log/2018-09.md]: Nope--GA in Preview for a month then GA in Prod is how most GAs are supposed to work (with major/minor release). We accidentally deleted it last release.

File: _source/_docs/api/resources/sessions.md [hunk 13]
```diff
@@ -564,6 +564,28 @@ If the session is invalid, a `404 Not Found` response will be returned.
...
+
+Example Request
+
+~~~http
+Accept: application/json
+Content-Type: application/json
+Authorization: SSWS ${api_token} or {sid_cookie}
+Prefer: return=minimal
+
+POST https://${yourDomain}.com/api/v1/sessions/{id-or-me}/refresh
```
Comment (tthompson-okta) [inline _source/_docs/api/resources/sessions.md]: Where did this URL come from? The [Refresh Session](https://developer.okta.com/docs/api/resources/sessions#refresh-current-session) endpoint should be `/api/v1/sessions/{id-or-me}/lifecycle/refresh`

File: _source/_docs/api/resources/sessions.md [hunk 14]
```diff
@@ -564,6 +564,28 @@ If the session is invalid, a `404 Not Found` response will be returned.
...
+
+Okta now supports [the HTTP Header `Prefer`](https://tools.ietf.org/html/rfc7240#section-4.2) in [the Sessions API for refreshing sessions](/docs/api/resources/sessions#refresh-current-session). You can extend the session lifetime, but skip any processing work related to building the response body.
+
+Example Request
+
+~~~http
+Accept: application/json
+Content-Type: application/json
+Authorization: SSWS ${api_token} or {sid_cookie}
+Prefer: return=minimal
```
Comment (tthompson-okta) [inline _source/_docs/api/resources/sessions.md]: This is correct, but I don't see it reflected at https://developer.okta.com/docs/api/resources/sessions#option-use-the-http-header-prefer -- is this just not published yet?

File: _source/_docs/api/resources/sessions.md [hunk 15]
```diff
@@ -564,6 +564,28 @@ If the session is invalid, a `404 Not Found` response will be returned.
 }
 ~~~
 
+#### Option: Use the HTTP Header Prefer
```
Comment (tthompson-okta) [inline _source/_docs/api/resources/sessions.md]: We want to encourage this as the default behavior (even though it's technically opt-in). So I think we should add `Prefer: return=minimal` header to all the examples in this doc:

* https://developer.okta.com/docs/api/resources/sessions#extend-session
* https://developer.okta.com/docs/api/resources/sessions#refresh-session
* https://developer.okta.com/docs/api/resources/sessions#refresh-current-session

and then update the response examples to show the new empty response:
```
HTTP/1.1 204 No Content
Preference-Applied: return=minimal
```
and then we could invert this "Option" section to describe basically how to get the response body back if you really need it (and the way to accomplish that would be to omit the `Prefer` header).

File: _source/_docs/api/resources/sessions.md (hunk 13 above)
Comment (mystiberry-okta) [reply _source/_docs/api/resources/sessions.md]: Likely a typo (accidental deletion), I'll fix.

File: _source/_docs/api/resources/sessions.md (hunk 14 above)
Comment (mystiberry-okta) [reply _source/_docs/api/resources/sessions.md]: Nothing pending, will fix.

File: _source/_docs/api/resources/sessions.md (hunk 15 above)
Comment (mystiberry-okta) [reply _source/_docs/api/resources/sessions.md]: I'll create a separate JIRA for this.

**Pull request merged**